File: C#/PBTKata/Bank/Account.cs
using System.Collections.Immutable;

namespace PBTKata.Bank
{
    public record Account(
        Amount Balance,
        bool IsOverdraftAuthorized,
        Amount MaxWithdrawal,
        ImmutableList<Withdraw>? Withdraws)
    {
        public ImmutableList<Withdraw> Withdraws { get; init; } = Withdraws ?? ImmutableList<Withdraw>.Empty;

        public Account Withdraw(Withdraw command)
        {
            return this with
            {
                Balance = Balance - command.Amount,
                Withdraws = Withdraws.Add(command)
            };
        }
    }
}

File: java/src/main/java/bank/Account.java
package bank;

import io.vavr.collection.Seq;

public record Account(
        double balance,
        boolean isOverdraftAuthorized,
        double maxWithdrawal,
        Seq<Withdraw> withdraws) {
    public Account withdraw(Withdraw command) {
        return new Account(balance - command.amount().value(),
                isOverdraftAuthorized,
                maxWithdrawal,
                withdraws().append(command));
    }
}